packages/fxa-settings/src/components/ThirdPartyAuth/index.tsx

```diff
@@ -2,20 +2,140 @@
 import { FtlMsg } from 'fxa-react/lib/utils';
+
 import { ReactComponent as GoogleLogo } from './google-logo.svg';
 import { ReactComponent as AppleLogo } from './apple-logo.svg';
 
+import { useAccount, useConfig, useStorage } from '../../models';
+import Storage from '../../lib/storage';
+
 export type ThirdPartyAuthProps = {
   enabled?: boolean;
 };
 
+type ThirdPartyAuthSigninParams = {
+  authorizationEndpoint: string;
+  clientId: string;
+  scope: string;
+  redirectUri: string;
+  responseType: string;
+  responseMode?: string;
+};
+
+const GOOGLE_SCOPES = 'openid email profile';
+const APPLE_SCOPES = 'email';
+
+/**
+ * ThirdPartyAuth component
+ * A React component that renders Google and Apple third-party authentication buttons.
+ * It handles user sign-in with the respective provider when the buttons are clicked.
+ */
 const ThirdPartyAuth = ({ enabled = false }: ThirdPartyAuthProps) => {
+  const config = useConfig();
+  const storageData = useStorage();
+  const account = useAccount();
+
+  useEffect(() => {
+    // TODO: Figure out why `storageData` is not available
+    const authParams = Storage.factory('localStorage', window).get(
+      'fxa_third_party_params'
+    );
+    if (authParams) {
+      completeSignIn();
+    }
+  }, []);
+
   if (!enabled) {
     return null;
   }
 
+  /**
+   * signIn - Handles the sign-in process for third-party authentication providers.
+   * Creates a form, sets necessary parameters, and submits it to the provider's
+   * authorization endpoint.
+   *
+   * @param {ThirdPartyAuthSigninParams} config - Configuration parameters for the signIn process.
+   */
+  function signIn(config: ThirdPartyAuthSigninParams) {
+    clearStoredParams();
+
+    // We stash originating location in the state oauth param
+    // because we will need it to use it to log the user into FxA
+    const currentParams = new URLSearchParams(window.location.search);
+    currentParams.delete('deeplink');
+    currentParams.set('showReactApp', 'true');
+
+    const state = encodeURIComponent(
+      `${window.location.origin}${
+        window.location.pathname
+      }?${currentParams.toString()}`
+    );
+
+    const form = window.document.createElement('form');
+
+    form.setAttribute('method', 'GET');
+    form.setAttribute('action', config.authorizationEndpoint);
+
+    const params: Record<string, string | undefined> = {
+      client_id: config.clientId,
+      scope: config.scope,
+      redirect_uri: config.redirectUri,
+      state,
+      access_type: 'offline',
+      prompt: 'consent',
+      response_type: config.responseType,
+      response_mode: config.responseMode,
+    };
+
+    for (const [key, value] of Object.entries(params)) {
+      if (!value) {
+        continue;
+      }
+      const input = window.document.createElement('input');
+      input.setAttribute('type', 'hidden');
+      input.setAttribute('name', key);
+      input.setAttribute('value', value);
```

Comment:
Q that I should probably know... When setting attributes dynamically, are they auto-escaped/sanitized, or do we need to guard against rouge quotes/chars?
Although none of these look like they are coming from the user (except maybe `state` which gets the current query string, but that's already URI encoded.